src/main/scala/rocket/CSR.scala

```diff
@@ -719,7 +760,16 @@ class CSRFile(
         reg_dcsr.prv := trimPrivilege(reg_mstatus.prv)
         new_prv := PRV.M
       }
-    }.elsewhen (delegate) {
+    }.elsewhen (trapToNmiInt) {
+      when (reg_rnmie || reg_unmie && causeIsUnmiInt) {
+        reg_rnmie := false.B
+        reg_unmie := !causeIsUnmiInt
+        reg_mnepc := epc
+        reg_mncause := (BigInt(1) << (xLen-1)).U | Mux(causeIsUnmiInt, 1.U, Mux(causeIsRnmiBEU, 3.U, 2.U))
+        reg_mnstatus.mpp := trimPrivilege(reg_mstatus.prv)
+        new_prv := PRV.M
+      }
+    }.elsewhen (delegate && nmie) {
```

Comment:
Why is `nmie` here?

Reply:
The draft RNMI spec says that exceptions that occur while an NMI handler is active are handled in M-mode and affect mepc, mcause, etc.  So they are never delegated.  The next case of this when/else is taken in that case.